uiServices/src/testCaseServices.go

```diff
@@ -188,4 +188,48 @@ func getAllTestCases(rw http.ResponseWriter, req *http.Request) {
 	}
 
 	rw.WriteHeader(http.StatusOK)
-}
\ No newline at end of file
+}
+
+func getTestCase(rw http.ResponseWriter, req *http.Request) {
+
+	testCasePathDir := getTestCaseHeader(req)
+	testCaseName := chi.URLParam(req, "testCaseName")
+
+	ValidateFileNameAndHeader(rw, req, testCasePathDir, testCaseName)
+
+	file, err := os.Open(fmt.Sprintf("%s%s.xml", testCasePathDir, testCaseName))
+	if err != nil {
+		logrus.Error("Test Case Name Not Found: " + testCasePathDir + testCaseName)
```

Comment:
Is this the correct log message? How do you know that "Test Case Name Not Found" while you are trying to open. In case `os.Open` fails for some other reason like file being open by another user or something similar will you not output wrong message?

Reply:
So will I add a check to see if file exists first, then try an os.Open? that will cover both scenarios.